qiskit/transpiler/passmanager.py

```diff
@@ -389,67 +389,66 @@ def __init__(self, stages=None, **kwargs):
         """
-        if stages is None:
-            self.stages = [
-                "init",
-                "layout",
-                "routing",
-                "translation",
-                "optimization",
-                "scheduling",
-            ]
-        else:
-            invalid_stages = [
-                stage for stage in stages if self.invalid_stage_regex.search(stage) is not None
-            ]
-            if invalid_stages:
-                with io.StringIO() as msg:
-                    msg.write(f"The following stage names are not valid: {invalid_stages[0]}")
-                    for invalid_stage in invalid_stages[1:]:
-                        msg.write(f", {invalid_stage}")
-                    raise ValueError(msg.getvalue())
-
-            self.stages = stages
+        stages = stages or [
+            "init",
+            "layout",
+            "routing",
+            "translation",
+            "optimization",
+            "scheduling",
+        ]
+        self._validate_stages(stages)
+        self._stages = stages
         super().__init__()
-        for stage in self.stages:
-            pre_stage = "pre_" + stage
-            post_stage = "post_" + stage
-            setattr(self, pre_stage, None)
-            setattr(self, stage, None)
-            setattr(self, post_stage, None)
-        for stage, pm in kwargs.items():
-            if (
-                stage not in self.stages
-                and not (stage.startswith("pre_") and stage[4:] in self.stages)
-                and not (stage.startswith("post_") and stage[5:] in self.stages)
-            ):
-                raise AttributeError(f"{stage} is not a valid stage.")
+        self._validate_kwargs(kwargs)
+        for stage in self.expanded_stages:
+            pm = kwargs.get(stage, None)
             setattr(self, stage, pm)
-        self._update_passmanager()
 
-    def _update_passmanager(self):
-        self._pass_sets = []
+    def _validate_stages(self, stages: List[str]) -> None:
+        invalid_stages = [
+            stage for stage in stages if self.invalid_stage_regex.search(stage) is not None
+        ]
+        if invalid_stages:
+            with io.StringIO() as msg:
+                msg.write(f"The following stage names are not valid: {invalid_stages[0]}")
+                for invalid_stage in invalid_stages[1:]:
+                    msg.write(f", {invalid_stage}")
+                raise ValueError(msg.getvalue())
+
+    def _validate_kwargs(self, kwargs: Dict[str, Any]) -> None:
+        for stage in kwargs.keys():
+            if stage not in self.expanded_stages:
+                raise AttributeError(f"{stage} is not a valid stage.")
+
+    @property
+    def stages(self) -> List[str]:
+        """Pass manager stages"""
+        return self._stages
+
+    @property
+    def expanded_stages(self) -> List[str]:
+        """Expanded Pass manager stages including 'pre_' and 'post_' phases."""
```

Comment:
The docs builds are failing because sphinx is trying to interpret this as a reference I think. Maybe try:

```suggestion
        """Expanded Pass manager stages including ``pre_`` and``'post_`` phases."""
```